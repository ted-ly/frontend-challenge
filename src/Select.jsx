import React from 'react';
import { toCapitalized } from './utils';

export default function Select({ options, name, ...otherProps }) {
    return (
        <select name={name} {...otherProps}>
            <option key='' value=''>Select Your Favorite Color</option>
            {(options || []).map(opt => <option key={opt} value={opt}>{toCapitalized(opt)}</option>)}
        </select>
    );
}