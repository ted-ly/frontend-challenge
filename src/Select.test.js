import renderer from 'react-test-renderer';
import Select from './Select';
import { shallow } from 'enzyme';

describe('Select', () => {
    it('should match snapshot', () => {
        const component = renderer.create(<Select />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with select options', () => {
        const component = renderer.create(<Select options={['black', 'white']} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should call stub function and change option', () => {
        const stubOnChange = jest.fn();
        const component = shallow(<Select options={['black', 'white']} onChange={stubOnChange} />);
        component.find('select').simulate('change', {target: {value: 'white'}});
        expect(stubOnChange.mock.calls).toHaveLength(1);
        expect(stubOnChange.mock.calls[0][0].target.value).toBe('white');
    })
});