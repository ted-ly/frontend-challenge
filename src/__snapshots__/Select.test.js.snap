// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
<select>
  <option
    value=""
  >
    Select Your Favorite Color
  </option>
</select>
`;

exports[`Select should match snapshot with select options 1`] = `
<select>
  <option
    value=""
  >
    Select Your Favorite Color
  </option>
  <option
    value="black"
  >
    Black
  </option>
  <option
    value="white"
  >
    White
  </option>
</select>
`;
